### PR TITLE
Big refactor for testability

### PR DESCRIPTION
**Pull request by swernli**
@jterry75 @jstarks 
Made quite a few changes to the internal structure of our HCS/HNS classes to make this module more unit testable in the future.
The changes are:
1) Add IHcs and IHns interfaces that all higher level callers use for making calls into the lower level services.
2) Add classes that implement the interfaces and make the actual calls required for functionality.
3) Update entrypoints to take an optional interface (where a fake or mock could be provided) and have them fallback to factory functions if none is provided.
4) Make factory functions and related classes for the wrapped methods public so that callers can directly use the HCS/HNS for more granular testing if desired.
5) Remove (by making private) the actual extern methods so that callers have to go through factories.

This also fixes an issue where the symbols produced for the compute virtualization product code are supposed to be marked portable.

### CODE REVIEW

File: src/Microsoft.Windows.ComputeVirtualization/Hcs.cs [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    public class HcsFactory
+    {
+        public static IHcs GetHcs()
+        {
+            return new Hcs();
+        }
+
+        private class Hcs : IHcs
+        {
+            public bool EnumerateComputeSystems(string query, out string computeSystems)
```
Comment (jterry75) [inline src/Microsoft.Windows.ComputeVirtualization/Hcs.cs]: Not a huge deal but you could alternatively implement these explicitly `bool IHcs.EnumerateComputeSystems(...)`

**Comment by jterry75** [comment]
I like it!

**Comment by swernli** [comment]
Any other feedback, or should I merge?

**Comment by jstarks** [comment]
LGTM

**Pull request merged**